src/services/socketIoServices.ts

```diff
@@ -0,0 +1,169 @@
+'use client'
+
+import { io } from 'socket.io-client'
+import { useEffect, useState } from 'react'
+
+export type ListenerRes = {
+	messageServer: string
+	roomIdServer: string
+	userNameServer: string
+}
+
+export interface Event extends ListenerRes {
+	timeStamp: string
+}
+
+const socket = io()
+
+export function socketRoomEmitter(
+	emitterName: string,
+	message: string,
+	roomId: string,
+	userName: string,
+) {
+	socket.emit(emitterName, message, roomId, userName)
+}
+
+export function useSocketListener(listenerName: string) {
+	const [listenerRes, setListenerRes] = useState<ListenerRes>({
+		messageServer: '',
+		roomIdServer: '',
+		userNameServer: '',
+	})
+	useEffect(() => {
+		function onListenerRes(
+			messageServer: string,
+			roomIdServer: string,
+			userNameServer: string,
+		) {
+			setListenerRes({ messageServer, roomIdServer, userNameServer })
+		}
+		socket.on(listenerName, onListenerRes)
+
+		return () => {
+			socket.off(listenerName, onListenerRes)
+		}
+	}, [listenerName])
+
+	return listenerRes
+}
+
+// Only use this listener for the SocketIoInfo component to display
+// events for testing purposes
+export function useListenerEvents(joinListener: string, dataListener: string) {
+	const [events, setEvents] = useState<Event[]>([])
+
+	useEffect(() => {
+		function onListenerResponse(
```

Comment:
question: This function seems specific just for 'user-connected', 'server-data', right? Or can it be used for any other listner? I don't think so... if so, then we could rename this to be more specific, like `useSocketInfo`

Reply:
Are you talking about the hook name on line 53 or the internal function on line 57? If the hook name, I would say leave it the same. Since it is specific to creating events to be used in the socket test/info component, I think the hook name should include the word events. And since it is a listener, that makes sense to be in the name. useListenerEvents describes what it does.

Reply:
I was referring to the hook name :) But all good!